pkg/codegen/nodejs/gen_program_test.go

```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
 	ex, err := executable.FindExecutable("yarn")
 	require.NoError(t, err, "Could not find yarn executable")
 	dir := filepath.Dir(path)
-	pkgs := nodejsPackages(dependencies)
+	pkgs := nodejsPackages(t, dependencies)
 	// We delete and regenerate package files for each run.
-	packageJSON := filepath.Join(dir, "package.json")
-	if err := os.Remove(packageJSON); !os.IsNotExist(err) {
-		require.NoError(t, err, "Failed to delete %s", packageJSON)
+	packageJSONPath := filepath.Join(dir, "package.json")
+	if err := os.Remove(packageJSONPath); !os.IsNotExist(err) {
+		require.NoError(t, err, "Failed to delete %s", packageJSONPath)
 	}
 	yarnLock := filepath.Join(dir, "yarn.lock")
 	if err := os.Remove(yarnLock); !os.IsNotExist(err) {
 		require.NoError(t, err, "Failed to delete %s", yarnLock)
 	}
 
-	err = integration.RunCommand(t, "link @pulumi/pulumi",
-		[]string{ex, "link", "@pulumi/pulumi"},
-		dir, &integration.ProgramTestOptions{})
-	require.NoError(t, err, "Failed to link @pulumi/pulumi")
-	for _, pkg := range pkgs {
-		err = integration.RunCommand(t, "yarn add and install",
-			[]string{ex, "add", pkg}, dir, &integration.ProgramTestOptions{})
-		require.NoError(t, err, "Could not install package: %q", pkg)
+	pkgInfo := npmPackage{
```

Comment:
I'm curious - why have you decided to change from yarn add commands to writing the package file explicitly?

Reply:
I think that the new form is more readable. We deal mostly with go types, so the code is cleaner to me. When I wrote the previous version, I didn't know that `npmPackage` existed.